sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/KeyVaultBackupClient.java

```diff
@@ -98,7 +176,119 @@ public String getVaultUrl() {
      */
     @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
     public SyncPoller<KeyVaultBackupOperation, String> beginBackup(String blobStorageUrl, String sasToken) {
-        return asyncClient.beginBackup(blobStorageUrl, sasToken).getSyncPoller();
+        if (blobStorageUrl == null) {
+            throw logger.logExceptionAsError(new NullPointerException(
+                String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                    "'blobStorageUrl'")));
+        }
+
+        if (sasToken == null) {
+            throw logger.logExceptionAsError(new NullPointerException(
+                String.format(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.PARAMETER_REQUIRED),
+                    "'sasToken'")));
+        }
+
+        Context context = Context.NONE;
+        return SyncPoller.createPoller(
+            getDefaultPollingInterval(),
+            cxt -> new PollResponse<>(LongRunningOperationStatus.NOT_STARTED, backupActivationOperation(blobStorageUrl, sasToken, context).apply(cxt)),
```

Comment:
Why do we need to add a lambda setting the status of the operation instead of simply calling `backupActivationOperation()`.

Reply:
because Poller expects a function that results in a PollResponse, this lambda transforms the activation result into a PollResponse.